weave/trace/weave_client.py

```diff
@@ -2111,4 +2196,71 @@ def elide_display_name(name: str) -> str:
     return name
 
 
-__docspec__ = [WeaveClient, Call, CallsIter]
+class ObjectVersionCollection:
+    """A collection of object versions for a single object.
+
+    This class provides access to all versions of a specific object.
+    By default, versions are sorted by version_index in ascending order
```

Comment:
Doesn't the constructor always sort?

Reply:
I believe the objects always come back in order.  This comment is just noting to users what the order is

Reply:
I found the comment confusing, maybe because it talks about "when created" but it sounds like you are talking about API response, and also this is something the user of the higher level APIs doesn't have control over.